src/main/java/pl/lukakan/showstracker/show/MovieService.java

```diff
@@ -0,0 +1,58 @@
+package pl.lukakan.showstracker.show;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import pl.lukakan.showstracker.cast.role.model.Role;
+import pl.lukakan.showstracker.show.model.Genre;
+import pl.lukakan.showstracker.show.model.Movie;
+import pl.lukakan.showstracker.show.repository.GenreRepository;
+import pl.lukakan.showstracker.show.repository.MovieRepository;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class MovieService {
+    private final MovieRepository movieRepository;
+    private final GenreRepository genreRepository;
+
+    @Autowired
+    public MovieService(MovieRepository movieRepository, GenreRepository genreRepository) {
+        this.movieRepository = movieRepository;
+        this.genreRepository = genreRepository;
+    }
+
+    public Optional<Movie> getById(Long id) {
+        return movieRepository.findById(id);
+    }
+
+    public void add(Movie movie) {
+        movie.getGenres().forEach(genre -> genre.removeMovie(movie));
+        movie.getGenres().forEach(genre -> genre.addMovie(movie));
+        movieRepository.save(movie);
+    }
+
+    public List<Role> getActors(Movie movie) {
+        return movie.getRoles().stream()
+                .filter(role -> role.getFunction().getName().equals("Actor"))
```

Comment:
Tutaj czegoś takiego raczej nie chcemy robić, czyli porównywać funkcję ze zhardcodowanym stringiem. Pomijając to, że jak będzie miał rolę "actor", albo "ACTOR" to też ci to nie zadziała, to praktycznie encja Function praktycznie nic nie wnosi do Twojej aplikacji, tylko utrudnia Ci życie. Jakbyś to function zrobił np. tak:

private Long id;
private String name;
private FunctionType functionType;

i FuntionType byłoby enumem z wartościami tego Reżysera i Aktora, to wtedy byś mógł sobie tutaj porównywać po wartościach z enuma, a przykładową funkcją byłoby

name: Aktor drugoplanowy
function: Aktor

Reply:
tak... ten fragment byl taki roboczy, zeby sprawdzic czy zadziała. Faktycznie enum lepiej pasuje. Dzieki!